### PR TITLE
[Issue 60] Bulk Upload for Import Data

### LINKED ISSUE #60: Backend/Frontend - Bulk Upload for Import Data
Users should be able to select multiple CSV files of the same type to be uploaded/parsed on this page: http://abalone.blrice.net/file_uploads/new.

_Acceptance Criteria:_
As a user, I can upload a maximum of 10 CSV files in bulk of the same type (e.g. Untagged Animal Assessment) from my computer.

### PR DESCRIPTION
**Pull request by colinsoleim**
### This PR:
- Resolves #60
- Changes the file_uploads#upload route to allow multiple files
- Adds controller tests for all routes in the FileUploadsController
- Adds spec/fixtures folder for easier file upload testing
- Deletes unused file_uploads_helper_spec file

### New Dependencies:
- Adds ['rails-controller-testing gem'](https://github.com/rails/rails-controller-testing), which allows the use of `assigns` in controller tests. There have been [long discussions](https://github.com/rails/rails/issues/18950) about how to test controllers and whether `assigns` should be used at all, so if people have other recommendations for controller testing, I'm happy to change my test design. This is just what I normally use on my own projects.

### Type of change

* New feature (non-breaking change which adds functionality)

### How Has This Been Tested?

- This feature has rspec tests and has been tested on various browsers. The current implementation uses the HTML <input multiple> tag which is supported by all major desktop browsers, and most mobile browsers. Here's a compatibility breakdown https://caniuse.com/#feat=input-file-multiple

### Screenshots

Screen Shot 2020-02-05 at 10 06 56 PM

Screen Shot 2020-02-05 at 10 07 00 PM

### CODE REVIEW

File: app/views/file_uploads/new.html.erb [hunk 1]
```diff
@@ -1,35 +1,23 @@
...
-              <span class="file-label">
-                Choose a file...
-              </span>
-            </span>
-            <span class="file-name" id="file-name">
-            </span>
-          <% end %>
+        <div>
+          <%= label_tag('input_file', 'Select up to 10 files to upload', class: 'label') %>
+          <%= file_field_tag 'input_files[]', multiple: true, required: true %>
```
Comment (colinsoleim) [inline app/views/file_uploads/new.html.erb]: I didn't see any support for multiple file upload in [Bulma's documentation](https://bulma.io/documentation/form/), so I just reverted back to the native file upload

Comment (mdworken) [reply app/views/file_uploads/new.html.erb]: 👍

File: app/controllers/file_uploads_controller.rb [hunk 2]
```diff
@@ -13,56 +20,50 @@ def new
...
-      record_class = @processed_file.category.constantize
-      @headers = [:validation_status].concat record_class::HEADERS.keys.map{|header| header.downcase}
-      records = []
-      errors_list = Array.new
-      full_path_file = Rails.root.join('storage', @processed_file.filename).to_s
+    @processed_file = ProcessedFile.find(params[:id])
+    record_class = @processed_file.category.constantize
+    @headers = [:validation_status] + record_class::HEADERS.keys.map(&:downcase)
+    records = []
+    errors_list = []
```
Comment (mdworken) [inline app/controllers/file_uploads_controller.rb]: 👍 for making it more consistent :)

File: spec/controllers/file_uploads_controller_spec.rb [hunk 3]
```diff
@@ -1,5 +1,89 @@
...
+
+      assigns[:file_uploads].each do |upload|
+        expect(upload.result_message).to include 'Successfully queued'
+      end
+    end
+
+    it 'should reject a CSV with incorrect headers' do
+      post :upload, params: invalid_header_file_params
+
+      expect(response.code).to eq '200'
```
Comment (mdworken) [inline spec/controllers/file_uploads_controller_spec.rb]: I could go either way on this, but should this be a 400 instead of a 200 maybe?

**Pull request merged**